File: FizzBuzz/FizzBuzz/src/FizzBuzz.java
public class FizzBuzz {
    public static void main(String[] args){
        for (int i = 1; i < 100; i++) {
            System.out.println(function(i));
        }
    }

    public static String function(int i) {
        if (i % 3 == 0 && i % 5 == 0) {
            return "FizzBuzz";
        }
        if (isFizz(i)) {
            return "Fizz";
        }
        if (isBuzz(i)) {
            return "Buzz";
        }
        return String.valueOf(i);
    }

    private static boolean isBuzz(int i) {
        return i % 5 == 0 || String.valueOf(i).indexOf('5')!=-1;
    }

    private static boolean isFizz(int i) {
        return i % 3 == 0|| String.valueOf(i).indexOf('3')!=-1;
    }
}

File: FizzBuzz/FizzBuzzKt/src/main/java/FizzBuzz.kt
object FizzBuzz {
    fun of(given: Int): String = when {
        given % 3 == 0 && given % 5 == 0  ->"FizzBuzz"
        given % 3 == 0 -> "Fizz"
        given % 5 == 0 -> "Buzz"
        else -> given.toString()
    }
}
